src/index.js

```diff
@@ -1,212 +1,169 @@
 import {visit} from 'unist-util-visit';
-import {JSDOM} from 'jsdom';
 import rangeParser from 'parse-numeric-range';
 import shiki from 'shiki';
-import sanitizeHtml from 'sanitize-html';
+import {unified} from 'unified';
+import rehypeParse from 'rehype-parse';
 
-// To make sure we only have one highlighter per theme in a process
+// Store only one highlighter per theme in a process
 const highlighterCache = new Map();
-
-function getThemesFromSettings(settings) {
-  if (
-    typeof settings.theme === 'string' ||
-    settings.theme?.hasOwnProperty('tokenColors')
-  ) {
-    return {default: settings.theme};
-  }
-
-  return settings.theme;
+const hastParser = unified().use(rehypeParse, {fragment: true});
+
+function toFragment({node, trees, lang}) {
+  node.tagName = 'span';
+  // User can replace this with a real Fragment at runtime
+  node.properties = {'data-rehype-pretty-code-fragment': ''};
+  node.children = Object.entries(trees).map(([mode, tree]) => {
+    const pre = tree.children[0];
+    // Remove class="shiki" and the background-color
+    pre.properties = {};
+
+    const code = pre.children[0];
+    code.properties['data-language'] = lang;
+    code.properties['data-theme'] = mode;
+
+    return pre;
+  });
 }
```

Comment:
This will enforce a wrapper fragment even if there is only a single  `theme`

cc: @renrizzolo